### PR TITLE
Replace `aws:username`, `jwt:` and `ldap:` policy variables in session policies

### PR DESCRIPTION
**Pull request by dvaldivia**
Fixes https://github.com/minio/console/issues/1803 
Fixes https://github.com/minio/console/issues/1287
Fixes https://github.com/minio/console/issues/1790
Fixes https://github.com/minio/console/issues/1597

Add support to replace `${aws:username}` for the current user username so the secure components can allow access to resoures with this variable in them

Signed-off-by: Daniel Valdivia <18384552+dvaldivia@users.noreply.github.com>

### CODE REVIEW

File: restapi/user_login.go [hunk 1]
```diff
@@ -88,15 +88,19 @@ func login(credentials ConsoleCredentialsI, sessionFeatures *auth.SessionFeature
...
 	if err != nil {
 		return nil, err
 	}
-	return iampolicy.ParseConfig(bytes.NewReader(accountInfo.Policy))
+	return &accountInfo, nil
+}
+
+// replacePolicyVariables replaces known variables from policies with known values
+func replacePolicyVariables(accountInfo *madmin.AccountInfo) json.RawMessage {
+	rawPolicy := bytes.ReplaceAll(accountInfo.Policy, []byte("${aws:username}"), []byte(accountInfo.AccountName))
```
Comment (harshavardhana) [inline restapi/user_login.go:102]: this is the wrong way to do this, you should extract the SessionToken - claims have all the keys needed to replace the values here.

**Review by harshavardhana — CHANGES_REQUESTED** [review]

File: restapi/user_login.go (hunk 1 above)
Comment (dvaldivia) [reply restapi/user_login.go:102]: that will come next, right now I went after a single variable, all the JWT related variables do need the STS extracted

Comment (harshavardhana) [reply restapi/user_login.go]: it should come in this PR alone, its a generic fix @dvaldivia you can't just do `aws:username` as one off, there is also `aws:userid` 

You need support to replace

```
aws:
jwt:
ldap:
```

And this is a generic fix.

Comment (dvaldivia) [reply restapi/user_login.go]: We don't have a `userid` equivalent, should that be access key instead?

I added `jwt:` and `ldap:` to this PR @harshavardhana

**Review by Alevsk — APPROVED** [review]

**Review by bexsoft — APPROVED** [review]

**Pull request merged**